draft-ietf-dnssd-srp-replication.xml

```diff
@@ -231,9 +231,17 @@
       <section anchor="startup">
         <name>Startup</name>
 
-        <t>When a partner starts SRP Replication, it enters the "startup" state and picks a random discovery time interval uniformly selected from the range [MIN_PARNTER_DISCOVERY_INTERVAL, MAX_PARTNER_DISCOVERY_INTERVAL]. The recommended minimum and maximum values are 4 and 7.5 seconds, respectively.</t>
+        <t>When a partner starts SRP Replication, it enters the "startup" state and picks a random discovery time interval uniformly selected from the range [MIN_PARNTER_DISCOVERY_INTERVAL, MAX_PARTNER_DISCOVERY_INTERVAL]. The recommended minimum and maximum values are 4 and 7.5 seconds, respectively. The new partner uses the discovery time to (1) discover and determine the preferred dataset ID, (2) find all existing SRPL partners, establish SRPL sessions with them and synchronize with them. The randomness introduced in the start time helps to handle the situation where a group of SRPL partners start around the same time, for example, triggered by a power outage.</t>
 
-        <t>When the discovery time interval expires, the SRPL partner transitions to the "routine operation" state (<xref target="routine-operation"/>) after it has either successfully synchronized with or NUM_DISCOVERY_SYNC_ATTEMPTS have occurred with each discovered partner. If the partner did not discover any other SRPL partners to synchronize with, it immediately transitions to the "routine operation" state.</t>
+        <t>When the discovery time interval expires, if the partner did not discover any other SRPL partners to synchronize with, it transitions to the "routine operation" state (<xref target="routine-operation"/>). If it discovers other SRPL partners, it can enter "routine operation" state after one of the following conditions are met for every partner it discovered during the discovery interval:</t>
+        <ul>
+          <li>The new partner has established an SRPL session with the discovered partner and successfully finished initial synchronization (<xref target="initial-synchronization"/>).</li>
+          <li>The new partner has failed to establish a connection and synchronize with the discovered partner (on any of its discovered addresses) for at least DISCOVERY_SYNC_TIMEOUT interval since the first connection attempt.  If the DISCOVERY_SYNC_TIMEOUT expires in the middle of an ongoing SRPL session establishment or initial synchronization with the discovered partner, the new partner MUST wait until the initial synchronization is either successfully completed or fails. The recommended value for DISCOVERY_SYNC_TIMEOUT is 5 seconds.</li>
```

Comment:
```suggestion
          <li>The new partner failed to establish a connection and synchronize any discovered partner (on any of their discovered addresses) for at least DISCOVERY_SYNC_TIMEOUT interval since the first connection attempt.  If the DISCOVERY_SYNC_TIMEOUT expires in the middle of an ongoing SRPL session establishment or initial synchronization with the discovered partner, the new partner MUST wait until the initial synchronization is either successfully completed or fails. The recommended value for DISCOVERY_SYNC_TIMEOUT is 5 seconds.</li>
```

Reply:
I know that it's traditional in e.g. Matter and Thread to do work like this in the Jira or in github, but for IETF work this stuff should be announced on the mailing list. I don't think it's a bad thing to open an issue, but not everybody in the working group will see it if you don't also mention it on the mailing list.

Reply:
Is it sufficient to post a link to this PR on the mailing list? Or should we post the proposed text on the mailing list so that we can discuss/edit there?

Reply:
Same here. This condition is applied per discovered partner (applying to each)